lm20/pipelines.py

```diff
@@ -170,12 +170,16 @@ def file_path(self, request, response=None, info=None, *, item=None):
 
         content_disposition = headers.get("Content-Disposition")
 
+        if type(content_disposition) is bytes:
```

Comment:
i think they might always be bytes?

Reply:
i believe this is where the `attributeerror` was coming from. at this point, most dispositions were coming in as strings, so the `decode` method raised that error. and same for the `content-type` it looks like, which is why i have another conditional near that

(caught this while fixing a variable assignment error that snuck past me)

Reply:
okay, cool.

generally, we prefer in python to not explicitly check for types. but when we do, we should use the isinstance method

```python
if isinstance(content_disposition, bytes):
...
```

one benefit of this is that it will be Truth for subclasses, which is usually handy.

but, genearlly, we want to just try to do the thing and catch an exception if it doesn't work so. for this we could do.

```python
try:
   content_disposition = content_disposition.encode()
except AttributeError:
   pass
```

Reply:
you can do this same pattern in the below section too.

Reply:
ah gotcha! so instead of checking for types, it's better to catch exceptions. and if we do feel like checking for a type is the way to go, stick to using `isinstance()`

Reply:
yes, the general idea is to not introduce an irrelevant brittleness.

if the try except pattern you don't actually care what type of thing "content_disposition".